internal/db/schema/migrations/oss/postgres/88/01_remove_session_connection_state.up.sql

```diff
@@ -0,0 +1,224 @@
+
+  drop trigger wh_insert_session_connection_state on session_connection_state;
+  drop function wh_insert_session_connection_state();
+
+  drop trigger wh_insert_session_connection on session_connection;
+  drop function wh_insert_session_connection();
+
+  --  If the connected_time_range is null, it means the connection is authorized but not connected.
+  --  If the upper value of connected_time_range is > now() (upper range is infinity) then the state is connected.
+  --  If the upper value of connected_time_range is <= now() then the connection is closed.
+  alter table session_connection
+    add column connected_time_range tstzrange;
+
+  -- Migrate existing data from session_connection_state to session_connection
+   update session_connection
+      set connected_time_range =
+  (select tstzrange(min(start_time), max(start_time))
+     from session_connection_state
+    where session_connection_state.connection_id = session_connection.public_id
+ group by connection_id );
+
+  drop table session_connection_state;
+  drop table session_connection_state_enm;
+
+  -- Insert on session_connection creates the connection entry, leaving the connected_time_range to null, indicating the connection is authorized
+  -- "Connected" is handled by the function ConnectConnection, which sets the connected_time_range lower bound to now() and upper bound to infinity
+  -- "Closed" is handled by the trigger function, update_connected_time_range_on_closed_reason, which sets the connected_time_range upper bound to now()
+  -- State transitions are guarded by the trigger function, check_connection_state_transition, which ensures that the state transitions are valid
+  create function check_connection_state_transition() returns trigger
+  as $$
+    begin
+    -- If old state was authorized, allow transition to connected or closed
+    if old.connected_time_range is null then
+      return new;
+    end if;
+
+    -- If old state was closed, no transitions are allowed
+    if upper(old.connected_time_range) < 'infinity' and old.connected_time_range != new.connected_time_range then
+      raise exception 'Invalid state transition from closed';
+    end if;
+
+    -- If old state was connected, allow transition to closed
+    if upper(old.connected_time_range) = 'infinity' and upper(new.connected_time_range) != 'infinity' then
+      return new;
+    else
+      raise exception 'Invalid state transition from connected';
+    end if;
+
+    return new;
+    end;
+  $$ language plpgsql;
+
+  create trigger check_connection_state_transition before update of connected_time_range on session_connection
+    for each row execute procedure check_connection_state_transition();
+
+  create function update_connected_time_range_on_closed_reason() returns trigger
+  as $$
+    begin
+      if new.closed_reason is not null then
+          if old.connected_time_range is null or old.connected_time_range = tstzrange(lower(old.connected_time_range), 'infinity'::timestamptz) then
```

Comment:
This conditional seems a little odd. In `check_connection_state_transition` we just compare the `upper` value, so I think the same check could be done with:

```suggestion
          if old.connected_time_range is null or upper(old.connected_time_range) = 'infinity'::timestamptz then
```

I think that is a little easier to read.